arches/app/views/workflow_history.py

```diff
@@ -0,0 +1,75 @@
+import json
+
+from django.db import transaction
+from django.db.models import F, JSONField, Value
+from django.db.models.expressions import CombinedExpression
+from django.db.utils import IntegrityError
+from django.utils.translation import gettext as _
+from django.views.generic import View
+
+from arches.app.utils.permission_backend import user_is_resource_reviewer
+from arches.app.utils.response import JSONErrorResponse, JSONResponse
+from arches.app.models import models
+
+
+class WorkflowHistoryView(View):
+
+    def get(self, request, workflowid):
+        if not user_is_resource_reviewer(request.user):
+            return JSONErrorResponse(_("Request Failed"), _("Permission Denied"), status=403)
```

Comment:
For AFS being a resource reviewer is required, but it's possible for other applications to have workflows that only require a user to be a resource editor. They would have to adjust their permissions, but it's possible. I think instead of this check, maybe we just use the group required decorator? ```@method_decorator(group_required("Resource Editor"), name="dispatch")``` above the get and post methods?

Reply:
Cool, I'll relax this to Resource Editor.

Thanks for showing me the decorators. They seem to return 302 (redirected to nowhere) instead of 403:
```py
======================================================================
FAIL: test_get_workflow_history (tests.views.workflow_tests.WorkflowHistoryTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jwalls/prj/arches/tests/views/workflow_tests.py", line 71, in test_get_workflow_history
    self.assertEqual(response.status_code, 403)
AssertionError: 302 != 403

======================================================================
FAIL: test_no_edits_after_completed (tests.views.workflow_tests.WorkflowHistoryTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jwalls/prj/arches/tests/views/workflow_tests.py", line 159, in test_no_edits_after_completed
    self.assertEqual(response.status_code, 403)
AssertionError: 302 != 403

======================================================================
FAIL: test_post_workflow_history (tests.views.workflow_tests.WorkflowHistoryTests)
Partial updates of componentdata and stepdata are allowed.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jwalls/prj/arches/tests/views/workflow_tests.py", line 85, in test_post_workflow_history
    self.assertEqual(response.status_code, 403)
AssertionError: 302 != 403

----------------------------------------------------------------------
Ran 5 tests in 0.103s
```

That also means no banner in the UI like this if we keep using a JSONErrorResponse:
<img width="1505" alt="Screenshot 2023-10-16 at 6 03 03 PM" src="https://github.com/archesproject/arches/assets/38668450/def1e3f9-9662-4cb2-8f46-40aa3ea262a2">


Which approach do you think is better?

Reply:
Ah - interesting. I think the 403 is better.

Reply:
👍 
relaxed to Resource Editor in [23c4e11](https://github.com/archesproject/arches/pull/10096/commits/23c4e114578ec730b4594aa04ab24ef1c3769301)